# PRAKTIKUM METODE NUMERIS Unit 5

## **LO4-Solusi persamaan**

Nama = Muchammad Hasan Chamdany                                      
NIM  = 20/456846/TK/50670

#**Simpson's 1/3 m Rule**

##Source Code :

In [38]:
import numpy as np
from scipy import integrate 

def f(x):
	return 8 + 4*np.cos(x)

def simpsons13m( a, b, n ):

	h = ( b - a )/n
	x = list()
	fx = list()
 
	i = 0
	while i<= n:
		x.append(a + i * h)
		fx.append(f(x[i]))
		i += 1

	result = 0
	i = 0
	while i<= n:
		if i == 0 or i == n:
			result+= fx[i]
		elif i % 2 != 0:
			result+= 4 * fx[i]
		else:
			result+= 2 * fx[i]
		i+= 1
	result = result * (h / 3)
	return result

print("Hasil Simpson's n = 2 --> %.5f"% simpsons13m(a = 0, b = np.pi/2, n=2))
print("Hasil Simpson's n = 4 --> %.5f"% simpsons13m(a = 0, b = np.pi/2, n=4))
print("True Value = 16.56637")

Hasil Simpson's n = 2 --> 16.57549
Hasil Simpson's n = 4 --> 16.56691
True Value = 16.56637


## Analisis :

dari hasil perhitungan menggunakan program Simpson's Rule 1/3 m yang telah dibuat, didapatkan bahwasannya ketika n lebih besar, maka akan semakin mendekati hasil dan nilai error nya semakin kecil. oleh karena itu, dengan menghitung secara analitik nilai dari True Value, didapat kesimpulan n = 4 lebih mendekati nilai dari True Value (16.56637).

#**Trapezoid**

##Source Code :

In [37]:
import numpy as np
from scipy import integrate

F = np.vectorize(f)

# n = 2
X2 = np.array([0, (np.pi/4), (np.pi/2)])
Y1 = F(X2)
# n = 4
X4 = np.array([0, (np.pi/8), (np.pi/4), (np.pi/2)])
Y2 = F(X4)

ResultN2 = np.trapz(Y1, X2)
ResultN4 = np.trapz(Y2, X4)

print("Trapezoid (n=2) -->",ResultN2)
print("Trapezoid (n=4) -->",ResultN4)

Trapezoid (n=2) --> 16.358608410233252
Trapezoid (n=4) --> 16.469076455635722


## Analisis

Dari hasil perhitungan menggunakan program Trapezoid dengan bantuan scipy.integrate  (numpy.trapz) didapatkan hasil Trapezoid (n=2) --> 16.358608410233252 dan
Trapezoid (n=4) --> 16.469076455635722. Sedangkan True Value = 16.56637. Dengan Demikian hasil yang lebih mendekati True Value adalah menggunakan Simpson's Rule dengan banyak segmen (n)

#**Kesimpulan**

Dapat disimpulkan pada praktikum Metode Numeris Unit 5 ini bahwasannya semakin banyak segmen yang dibentuk, errornya akan semakin kecil, sehingga hasil dari perhitungan tersebut akan semakin mendekati dengan True Value (nilai sebenarnya) dan didapatkan kesimpulan terhadap hasil yang di uji menggunakan metode Simpson's 1/3 Vs Trapezoid dengan hasil Simpson's 1/3 lebih mendekati True Value dibanding dengan Trapezoid